internal/backendconnection/otelcolresources/config.yaml.template

```diff
@@ -1,115 +1,238 @@
-        endpoint: {{ .IngressEndpoint }}
+  otlphttp:
+    endpoint: {{ .IngressEndpoint }}
 {{- if .HasExportAuthentication }}
-        headers:
-            Authorization: Bearer ${env:AUTH_TOKEN}
+    headers:
+      Authorization: Bearer ${env:AUTH_TOKEN}
 {{- end }}
 {{- end }}
 
 extensions:
-    health_check:
-        endpoint: ${env:MY_POD_IP}:13133
+  health_check:
+    endpoint: ${env:MY_POD_IP}:13133
+  file_storage/filelogreceiver_offsets:
+    directory: /var/otelcol/filelogreceiver_offsets
+    timeout: 1s
 
 processors:
-    batch: {}
-    k8sattributes:
-        extract:
-            metadata:
-            - k8s.namespace.name
-            - k8s.deployment.name
-            - k8s.statefulset.name
-            - k8s.daemonset.name
-            - k8s.cronjob.name
-            - k8s.job.name
-            - k8s.node.name
-            - k8s.pod.name
-            - k8s.pod.uid
-            - k8s.pod.start_time
-        filter:
-            node_from_env_var: K8S_NODE_NAME
-        passthrough: false
-        pod_association:
-        - sources:
-            - from: resource_attribute
-              name: k8s.pod.ip
-        - sources:
-            - from: resource_attribute
-              name: k8s.pod.uid
-        - sources:
-            - from: connection
-    memory_limiter:
-        check_interval: 5s
-        limit_percentage: 80
-        spike_limit_percentage: 25
+  batch: {}
+  filter/only_dash0_monitored_resources:
+    error_mode: ignore
+    traces:
+      span:
+      - 'resource.attributes["dash0.monitoring.instrumented"] != "true"'
+    metrics:
+      metric:
+      datapoint:
+      - 'resource.attributes["dash0.monitoring.instrumented"] != "true"'
+    logs:
+      log_record:
+      - 'resource.attributes["dash0.monitoring.instrumented"] != "true"'
+  k8sattributes:
+    extract:
+      metadata:
+      - k8s.namespace.name
+      - k8s.deployment.name
+      - k8s.statefulset.name
+      - k8s.daemonset.name
+      - k8s.cronjob.name
+      - k8s.job.name
+      - k8s.node.name
+      - k8s.pod.name
+      - k8s.pod.uid
+      - k8s.pod.start_time
+      labels:
+      - key: dash0.com/instrumented
+        tag_name: dash0.monitoring.instrumented
+        from: pod
+    filter:
+      node_from_env_var: K8S_NODE_NAME
+    passthrough: false
+    pod_association:
+    - sources:
+      - from: resource_attribute
+        name: k8s.pod.ip
+    - sources:
+      - from: resource_attribute
+        name: k8s.pod.uid
+    - sources:
+      - from: connection
+  memory_limiter:
+    check_interval: 5s
+    limit_percentage: 80
+    spike_limit_percentage: 25
 
 receivers:
-    otlp:
-        protocols:
-            grpc:
-                endpoint: ${env:MY_POD_IP}:4317
-            http:
-                endpoint: ${env:MY_POD_IP}:4318
+  otlp:
+    protocols:
+      grpc:
+        endpoint: ${env:MY_POD_IP}:4317
+      http:
+        endpoint: ${env:MY_POD_IP}:4318
+  # TODO Turn on conditionally for monitored namespaces
```

Comment:
Same here, is this TODO something that needs to be taken care of? The way it is now I think the TODO would end up in config maps in the customer's systems, which is not good for optics IMO.